src/services/external/transaction.service.ts

```diff
@@ -0,0 +1,48 @@
+import { Resource } from "@companieshouse/api-sdk-node";
+import ApiClient from "@companieshouse/api-sdk-node/dist/client";
+import { logger } from "../../utils/logger";
+import { Transaction } from "@companieshouse/api-sdk-node/dist/services/transaction/types";
+
+
+export class TransactionService {
+    constructor(private apiClient: ApiClient) {}
+
+    async postTransactionRecord(companyNumber: string, reference: string, description: string): Promise<Transaction> {
+        const transactionRecord = this.createTransactionRecord(companyNumber, reference, description);
+
+        logger.debug(`Created Transaction Record using Company Number: ${companyNumber}, Reference: ${reference}, Description: ${description}`);
+
+        const transactionService = this.apiClient.transaction;
+        const transactionServiceResponse = await transactionService.postTransaction(transactionRecord);
+
+        if (transactionServiceResponse.httpStatusCode !== 201) {
+            logger.error(`Non 201 response from transaction service. ${companyNumber}`);
+            throw transactionServiceResponse;
+        }
+
+        if (!isResource(transactionServiceResponse)) {
+            logger.error(`Transaction service response did not include a resource. Response: ${companyNumber}`);
+            throw new Error("Transaction service didn't return a resource");
+        }
+
+        if(transactionServiceResponse.resource == undefined){
```

Comment:
Could we include this check in the isResource function below, reduce the lines of code in the function